Yoast/yoast-components
# Implement Snippet Preview

## Summary

This PR can be summarized in the following changelog entry:

* Add Snippet Preview component

## Relevant technical choices:

This is only the snippet preview and not the editor. It supports the following elements:

* Title.
* URL.
* Description.

With all the features the current snippet preview has in the plugin:

* Highlighting of keyword in URL & description with transliteration.
* Changing the color based on whether the description is generated.
* onClick prop all three elements.
* Hover and active carets on all three elements.
* Show scroll hint to user if the screen is too small for the preview.

## Test instructions

This PR can be tested by following these steps:

* Start yoast-components using `yarn start`.
* Open the current snippet preview in Yoast SEO.
* Make sure they work the same.
